beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -394,24 +395,24 @@ private SafeFuture<Map<UInt64, List<Attestation>>> getAttestationsIncludedInEpoc
   }
 
   @Override
-  public void saveProducedAttestation(Attestation attestation) {
-    UInt64 epoch = spec.computeEpochAtSlot(attestation.getData().getSlot());
-    Set<Attestation> attestationsInEpoch =
+  public void saveProducedAttestation(final Attestation attestation) {
+    final UInt64 epoch = spec.computeEpochAtSlot(attestation.getData().getSlot());
+    final Set<Attestation> attestationsInEpoch =
         producedAttestationsByEpoch.computeIfAbsent(epoch, __ -> concurrentSet());
     attestationsInEpoch.add(attestation);
   }
 
   @Override
-  public void saveProducedBlock(SignedBeaconBlock block) {
-    UInt64 epoch = spec.computeEpochAtSlot(block.getSlot());
-    Set<SlotAndBlockRoot> blocksInEpoch =
+  public void saveProducedBlock(final SignedBlockContainer blockContainer) {
+    final UInt64 epoch = spec.computeEpochAtSlot(blockContainer.getSlot());
+    final Set<SlotAndBlockRoot> blocksInEpoch =
         producedBlocksByEpoch.computeIfAbsent(epoch, __ -> concurrentSet());
-    blocksInEpoch.add(new SlotAndBlockRoot(block.getSlot(), block.getRoot()));
+    blocksInEpoch.add(new SlotAndBlockRoot(blockContainer.getSlot(), blockContainer.getRoot()));
```

Comment:
I added `getSlotAndBlockRoot()` deep in the block hierarchy. Might be good to expose it via container too. Or simply use `getSignedBlock().getSlotAndBlockRoot()` as you prefer

Reply:
good idea

Reply:
Btw tried that but there was a clash with MinimalBeaconSummary default method, which is the same, so removed it and just changed it to `blockContainer.getSignedBlock().getSlotAndBlockRoot()` in this class